public/docs/ts/latest/tutorial/toh-pt6.jade

```diff
@@ -0,0 +1,252 @@
+
+:marked
+  In order to differentiate between add and edit we are adding a check to see if an id is passed in the url. If the id is absent we bind `HeroDetailComponent` to an empty `Hero` object. In either case, any edits made through the UI will be bound back to the same `hero` property.
+
+  The next step is to add a save method to `HeroDetailComponent` and call the corresponding save method in `HeroesService`.
+  
++makeExample('toh-6/ts/app/hero-detail.component.ts', 'save', 'app/hero-detail.component.ts (save)')(format=".")  
+
+:marked
+  The same save method is used for both add end edit since `HeroService` will know when to call `post` vs `put` based on the state of the `Hero` object.
+  
+  We mentioned earlier that `save()` returns a promise, so when the promise resolves, we call `emit` to notify `HeroesComponent` that we just added or modified a hero. `HeroesComponent` is listening for this notification and will automatically refresh the list of heroes to include our recent updates.     
+
+.l-sub-section
+  :marked
+    The `emit` "handshake" between `HeroDetailComponent` and `HeroesComponent` is an example of component to component communication. This is a topic for another day, but we have detailed information in our <a href="/docs/ts/latest/cookbook/component-communication.html#!#child-to-parent">Component Interaction Cookbook</a>  
+
+:marked
+  Here is `HeroDetailComponent` with the added save button.
+  
+figure.image-display
+  img(src='/resources/images/devguide/toh/hero-details-save-button.png' alt="Hero Details With Save Button") 
+
+:marked
+  ### Delete
+  
+  We have added the option to delete hereos from `HeroesComponent`. `HeroService` will delete the hero, but we still have to filter out the deleted hero from the list to update the view.
+  
++makeExample('toh-6/ts/app/heroes.component.ts', 'delete', 'app/heroes.component.ts (delete)')(format=".")  
+
+:marked
+  Here is `HeroesComponent` with the delete button. 
+  
+figure.image-display
+  img(src='/resources/images/devguide/toh/heroes-list-delete-button.png' alt="Heroes List With Delete Button")  
+
+.l-main-section  
+:marked
+ ## Integration with Http
+  
+  `Http` is not part of Angular core, but exists as a separate add-on module called `angular2/http`. It ships as a separate bundle, so we have to include the bundle in `index.html`.
+  
++makeExample('toh-6/ts/index.html', 'http', 'index.html (http)')(format=".")  
+  
+:marked
+  The only other thing we have to do is import `HTTP_PROVIDERS` from `angular2/http` and register it in `AppComponent`'s provider array.   
+  
++makeExample('toh-6/ts/app/app.component.ts', 'http', 'app/app.component (http)')(format=".")  
+
+:marked
+  ### Simulating the api
+  Our hero api is not deployed to production yet, but for the purposes of this chapter we have decided to integrate with an in-memory web api simulator.
```

Comment:
Ah, i see you are using the in memory web api ... cool. I think perhaps it may be better not to call it a pure database above though ... and instead do the call out to it earlier. Reading it, i was confused on where the DB came from.